dev-util/capstone/patches/capstone-4.0.1.patchset

```diff
@@ -0,0 +1,80 @@
+From 203f796d921948ea60d9941f4a19317b41d66dbe Mon Sep 17 00:00:00 2001
+From: Calvin Hill <calvin@hakobaito.co.uk>
+Date: Sat, 12 Jan 2019 22:54:57 +0000
+Subject: update patches for 4.0.1
+
+
+diff --git a/Makefile b/Makefile
+index 378f2fe..460baea 100644
+--- a/Makefile
++++ b/Makefile
+@@ -65,11 +65,11 @@ else
+ BLDIR = $(abspath $(BUILDDIR))
+ OBJDIR = $(BLDIR)/obj
+ endif
+-INCDIR ?= $(PREFIX)/include
++INCDIR ?= $(DESTDIR)$(PREFIX)/develop/headers
+ 
+ UNAME_S := $(shell uname -s)
+ 
+-LIBDIRARCH ?= lib
++LIBDIRARCH ?= lib/x86
+ # Uncomment the below line to installs x86_64 libs to lib64/ directory.
+ # Or better, pass 'LIBDIRARCH=lib64' to 'make install/uninstall' via 'make.sh'.
+ #LIBDIRARCH ?= lib64
+@@ -513,6 +513,7 @@ define generate-pkgcfg
+ 	echo 'Name: capstone' > $(PKGCFGF)
+ 	echo 'Description: Capstone disassembly engine' >> $(PKGCFGF)
+ 	echo 'Version: $(PKG_VERSION)' >> $(PKGCFGF)
++	echo 'prefix=$(PREFIX)' >> $(PKGCFGF)
+ 	echo 'libdir=$(LIBDIR)' >> $(PKGCFGF)
+ 	echo 'includedir=$(INCDIR)' >> $(PKGCFGF)
+ 	echo 'archive=$${libdir}/libcapstone.a' >> $(PKGCFGF)
+-- 
+2.19.1
+
+
+From 12215d491664fab5e11d359edc70dcda513cc1f2 Mon Sep 17 00:00:00 2001
+From: Calvin Hill <calvin@hakobaito.co.uk>
+Date: Sun, 13 Jan 2019 00:39:39 +0000
+Subject: Fix build of fuzztests.
+
+
+diff --git a/suite/fuzz/driverbin.c b/suite/fuzz/driverbin.c
+index 57eea2c..280a7a0 100644
+--- a/suite/fuzz/driverbin.c
++++ b/suite/fuzz/driverbin.c
+@@ -3,6 +3,7 @@
+ #include <stdio.h>
+ #include <dirent.h>
+ #include <unistd.h>
++#include <sys/stat.h>
+ 
+ int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size);
+ 
+@@ -13,6 +14,8 @@ int main(int argc, char** argv)
+     size_t Size;
+     DIR *d;
+     struct dirent *dir;
++    struct stat s;
++
+     int r = 0;
+ 
+     if (argc != 2) {
+@@ -32,9 +35,10 @@ int main(int argc, char** argv)
+ 
+     while((dir = readdir(d)) != NULL) {
+         //opens the file, get its size, and reads it into a buffer
+-        if (dir->d_type != DT_REG) {
+-            continue;
+-        }
++	stat(dir->d_name, &s);
++	if(!S_ISREG(s.st_mode)){
++	   continue;
++	}
+         //printf("Running %s\n", dir->d_name);
+         fp = fopen(dir->d_name, "rb");
+         if (fp == NULL) {
+-- 
+2.19.1
+
```

Comment:
Maybe this looks a "bit" better and could be upstreamed?
```
diff --git a/suite/fuzz/driverbin.c b/suite/fuzz/driverbin.c
index 57eea2c..65a046c 100644
--- a/suite/fuzz/driverbin.c
+++ b/suite/fuzz/driverbin.c
@@ -3,6 +3,7 @@
 #include <stdio.h>
 #include <dirent.h>
 #include <unistd.h>
+#include <sys/stat.h>
 
 int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size);
 
@@ -13,6 +14,8 @@ int main(int argc, char** argv)
     size_t Size;
     DIR *d;
     struct dirent *dir;
+    struct stat s;
+
     int r = 0;
 
     if (argc != 2) {
@@ -32,9 +35,16 @@ int main(int argc, char** argv)
 
     while((dir = readdir(d)) != NULL) {
         //opens the file, get its size, and reads it into a buffer
+        #ifndef __HAIKU__
         if (dir->d_type != DT_REG) {
             continue;
         }
+        else
+        stat(dir->d_name, &s);
+        if(!S_ISREG(s.st_mode)){
+        continue;
+        }
+        #endif
         //printf("Running %s\n", dir->d_name);
         fp = fopen(dir->d_name, "rb");
         if (fp == NULL) {
-- 
2.19.1```

Reply:
Applied suggested changes into patchset.